vectorstores/alloydb/README.md

```diff
@@ -0,0 +1,147 @@
+- **Improved metadata handling**: store metadata in columns instead of JSON, resulting in significant performance improvements.
+- **Clear separation**: clearly separate table and extension creation, allowing for distinct permissions and streamlined workflows.
+- **Better integration with AlloyDB**: built-in methods to take advantage of AlloyDB's advanced indexing and scalability capabilities.
+
+## Quick Start
+
+In order to use this package, you first need to go through the following
+steps:
+
+1. [Select or create a Cloud Platform project.](https://console.cloud.google.com/project)
+2. [Enable billing for your project.](https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project)
+3. [Enable the AlloyDB API.](https://console.cloud.google.com/flows/enableapi?apiid=alloydb.googleapis.com)
+4. [Authentication with CloudSDK.](https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login)
+
+## Supported Go Versions
+
+Go version >= go 1.22.0
+
+## Engine Creation
+
+The `AlloyDBEngine` configures a connection pool to your AlloyDB database. 
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+
+  "github.com/tmc/langchaingo/internal/alloydbutil"
+)
+
+func NewAlloyDBEngine(ctx context.Context) (*alloydbutil.PostgresEngine, error) {
+	// Call NewPostgresEngine to initialize the database connection
+    pgEngine, err := alloydbutil.NewPostgresEngine(ctx,
+        alloydbutil.WithUser("my-user"),
+        alloydbutil.WithPassword("my-password"),
+        alloydbutil.WithDatabase("my-database"),
+        alloydbutil.WithAlloyDBInstance("my-project-id", "region", "my-cluster", "my-instance"),
+    )
+    if err != nil {
+        return nil, fmt.Errorf("Error creating PostgresEngine: %s", err)
+    }
+    return pgEngine, nil
+}
+
+func main() {
+    ctx := context.Background()
+    alloyDBEngine, err := NewAlloyDBEngine(ctx)
+    if err != nil {
+         return nil, err
+    }
+}
+```
+
+See the full [Vector Store example and tutorial](https://github.com/tmc/langchaingo/tree/main/examples/google-alloydb-vectorstore-example).
+
+## Engine Creation WithPool
+
+Create an AlloyDBEngine with the `WithPool` method to connect to an instance of AlloyDB Omni or to customize your connection pool.
+
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+
+  "github.com/jackc/pgx/v5/pgxpool"
+  "github.com/tmc/langchaingo/internal/alloydbutil"
+)
+
+func NewAlloyDBWithPoolEngine(ctx context.Context) (*alloydbutil.PostgresEngine, error) {
+    myPool, err := pgxpool.New(ctx, os.Getenv("DATABASE_URL"))
+    if err != nil {
+        return err
+    }
+	// Call NewPostgresEngine to initialize the database connection
+    pgEngineWithPool, err := alloydbutil.NewPostgresEngine(ctx,
+        alloydbutil.WithUser("my-user"),
+        alloydbutil.WithPassword("my-password"),
+        alloydbutil.WithDatabase("my-database"),
```

Comment:
These shouldn't be necessary using WithPool, correct?

Reply:
The current design do request user and password or I Am Account email in the getUser method: https://github.com/averikitsch/langchaingo/blob/main/internal/alloydbutil/engine.go#L93

Reply:
made the changes in the engine to accept pool or the psswrd/user/database